### PR TITLE
Adds an admin spawnable Stunbaton variant, the SCAM boomerang.

### PR DESCRIPTION
**Pull request by ArcaneMusic**
## About The Pull Request

This started out as a meme April Fools PR to fuck with stunbatons, and then ATH convinced me to make it a real thing, and then I spent the last 2 days fucking with Git and downloading the source code over about 9 fucking times, now it's here.

Functionally acts as love child between bananium shields and the default stunbaton, complete with a default shitty battery to serve as a one-off ranged stun before needing a charge unless it's upgraded with a less shit battery.  Inherits all the flaws of the default stun baton, and requires manually catching them upon throw to make them more challenging and reflex based to balance the potential of having a spammable ranged stun.

(SCAM stands for Security Critical Aeromation Matrices, or, what you'll say when you hit yourself the first time with one.)

## Why It's Good For The Game

They're admin spawnable, so no harm no fowl. They're also a fun variant of the stun baton with the potential drawback of hitting yourself if thrown.

## Changelog
:cl: Arcane
add: Australian Justice can now be dished out.
/:cl:

### CODE REVIEW

**Comment by Tlaltecuhtli** [comment]
make it a premium item from sec vendor, maybe with less cell power to max 1 hit

**Comment by Borisvanmemes** [comment]
Add it to the sec vendor, it looks REALLY fun, if a tad powerful.

**Comment by knacker48** [comment]
How much should it cost?

**Comment by ATH1909** [comment]
> Add it to the sec vendor, it looks REALLY fun, if a tad powerful.

This, but make the ones in the sec vendors not come back after you throw them.

Alternatively, maybe make throwing a normal stunbaton while you know krav maga boost the stunbaton's thrown hit chance up to 100% (for that throw), since krav maga is kind of mostly redundant with stunbatons (or the other way around, depending on your point of view) anyway.

I really don't like the 99% chance to hit (instead of a 100% chance to hit). People already get mad about someone's riot shield blocking their attacks three times in a row (a 12.5% chance); I can only see a 1% chance to avoid being stunned by an item that should have hit you infuriating many people and causing a lot of salt.

But yeah, I do support this PR and will likely TC trade for this item+instant summons at some point if it gets merged.

File: code/game/objects/items/stunbaton.dm [hunk 1]
```diff
@@ -205,3 +205,58 @@
...
+	item_state = "boomerang"
+	force = 5
+	throwforce = 5
+	throw_range = 5
+	stunforce = 100
+	hitcost = 2000
+	throw_hit_chance = 99  //Have you prayed today?
+
+/obj/item/melee/baton/scam/Initialize()
+	. = ..()
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/objects/items/stunbaton.dm]: purpose?

File: code/game/objects/items/stunbaton.dm [hunk 2]
```diff
@@ -205,3 +205,58 @@
...
+
+/obj/item/melee/baton/scam/attack_self(mob/living/carbon/human/user)
+	..()
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/objects/items/stunbaton.dm]: purpose?

File: code/game/objects/items/stunbaton.dm [hunk 3]
```diff
@@ -205,3 +205,58 @@
...
+			if(status==1 && prob(throw_hit_chance) && iscarbon(hit_atom))
+				baton_stun(hit_atom)
+		if(thrownby && !caught)
+			sleep(1)
+			throw_at(thrownby, throw_range+2, throw_speed, null, TRUE)
+	else
+		return ..()
+
+/obj/item/melee/baton/scam/Initialize()
+	. = ..()
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/objects/items/stunbaton.dm]: purpose?

Comment (ExcessiveUseOfCobblestone) [reply code/game/objects/items/stunbaton.dm]: duplicate anyways

File: code/game/objects/items/stunbaton.dm [hunk 4]
```diff
@@ -205,3 +205,58 @@
...
+
+/obj/item/melee/baton/scam/loaded //this one starts with a cell pre-installed.
+	preload_cell_type = /obj/item/stock_parts/cell/upgraded
+
+
+/obj/item/melee/baton/scam/update_icon()
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/objects/items/stunbaton.dm]: put further subtypes below the procs that you're editing

**Comment by Barhandar** [comment]
Stunahawks _completely_ miss the point of boomerang stunbatons (that is, punishment for missing).

**Comment by ATH1909** [comment]
> Stunahawks _completely_ miss the point of boomerang stunbatons (that is, punishment for missing).

The punishment for missing is that you've just tossed your intended target a potent stun weapon that they will likely run off with and keep forever and/or use against you.

**Comment by ATH1909** [comment]
Somewhat relevant: https://www.youtube.com/watch?v=rZdEZSsIZIs

**Review by SpaceManiac — CHANGES_REQUESTED** [review]
what Cobby said

**Comment by ArcaneMusic** [comment]
Hopefully fixed all the style changes and removed the Duplicate Procs.

Gitbash scares me.

**Comment by Jalleo** [comment]
What happens when it had a bluespace power cell filled with plasma when its used? Does it explode? Please tell me it does.

**Comment by ArcaneMusic** [comment]
Explodes on contact with the first target hit. This includes upon self-stun.

**Review by ArcaneMusic — COMMENTED** [review]
Cleaned everything up, removed duplicate and blank methods as well as moved the subtype.

File: code/game/objects/items/stunbaton.dm [hunk 5]
```diff
@@ -205,3 +205,47 @@
...
+	return ..()
+
+/obj/item/melee/baton/scam/throw_impact(atom/hit_atom, datum/thrownthing/throwingdatum)
+	if(status==1)
+		var/caught = hit_atom.hitby(src, FALSE, FALSE, throwingdatum=throwingdatum)
+		if(iscarbon(hit_atom) && !caught)//if they are a carbon and they didn't catch it
+			if(status==1 && prob(throw_hit_chance) && iscarbon(hit_atom))
+				baton_stun(hit_atom)
+		if(thrownby && !caught)
+			sleep(1)
```
Comment (optimumtact) [inline code/game/objects/items/stunbaton.dm]: after you sleep you have to check !qdeleted(src)

**Pull request closed without merging**